projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/user/entity/User.kt

```diff
@@ -14,9 +14,146 @@ class User(
     @Column(name = "user_id")
     val id: Long,
 
-    @Column(name = "forename")
+    @ManyToOne
+    @JoinColumn(name = "staff_id")
+    val staff: Staff? = null,
+
+    @Column(name = "distinguished_name")
+    val username: String,
+
+    @Column
     val forename: String,
 
-    @Column(name = "surname")
-    val surname: String
-)
\ No newline at end of file
+    @Column
+    val surname: String,
+)
+
+@Immutable
+@Entity
+@Table(name = "staff")
+class Staff(
+
+    @Column(name = "officer_code", columnDefinition = "char(7)")
+    val code: String,
+
+    @Column
+    val forename: String,
+
+    @Column
+    val surname: String,
+
+    @JoinColumn(name = "probation_area_id")
+    @ManyToOne
+    val provider: Provider,
+
+    @OneToMany(mappedBy = "staff")
+    val caseLoad: List<Caseload> = emptyList(),
+
+    @ManyToMany
+    @JoinTable(
+        name = "staff_team",
+        joinColumns = [JoinColumn(name = "staff_id")],
+        inverseJoinColumns = [JoinColumn(name = "team_id")]
+    )
+    val teams: List<Team> = emptyList(),
+
+    @Id
+    @Column(name = "staff_id")
+    val id: Long
+) {
+    fun isUnallocated(): Boolean {
+        return code.endsWith("U")
+    }
+}
+
+@Entity
+@Immutable
+@Table(name = "team")
+class Team(
+    @Id
+    @Column(name = "team_id")
+    val id: Long,
+
+    @Column(name = "code", columnDefinition = "char(6)")
+    val code: String,
+
+    @Column(name = "description")
+    val description: String,
+
+    @ManyToMany
+    @JoinTable(
+        name = "staff_team",
+        joinColumns = [JoinColumn(name = "team_id")],
+        inverseJoinColumns = [JoinColumn(name = "staff_id")]
+    )
+    val staff: List<Staff>
+)
+
+@Immutable
+@Entity
+@Table(name = "probation_area")
+class Provider(
+    @Column(name = "code", columnDefinition = "char(3)")
+    val code: String,
+
+    val description: String,
+
+    @Id
+    @Column(name = "probation_area_id")
+    val id: Long,
+
+    @Column(name = "end_date")
+    val endDate: LocalDate? = null
+)
+
+@Entity
+@Immutable
+@Table(name = "case_load")
```

Comment:
Is this correct?  I think the table name is `caseload`.

Also, I wonder if we should be using the caseload table at all?  I know we've had issues in the past where it hasn't been kept up-to-date with the source tables (`offender_manager`, `order_manager` etc), but it is what the Delius caseload diary uses...

Reply:
ive corrected the table name to caseload. Ive also added an SQL restriction for now to only return role_code = 'OM'